bigquery_etl/shredder/bug1751979.py

```diff
@@ -249,3 +249,119 @@
+) AS (
+  (
+    WITH base AS (
+      SELECT
+        -- This reverses the separation logic in main_summary where
+        -- engine and source are parsed from the search_counts key
+        CONCAT(engine, '.', source) AS key,
+        count
+      FROM
+        UNNEST(input)
+    ),
+    parsed AS (
+      SELECT
+        *,
+        REGEXP_EXTRACT(key, "([^.]+[.]in-content[:.][^:]+:).*") AS prefix,
+        REGEXP_EXTRACT(key, "[^.]+[.]in-content[:.][^:]+:(.*)") AS code,
+      FROM
+        base
+    ),
+    scrubbed AS (
+      SELECT
+        * REPLACE (
+          IF(
+            prefix IS NULL
+            OR code IN (
+              "none",
+              "other",
+              "hz",
+              "h_",
+              "MOZ2",
+              "MOZ4",
+              "MOZ5",
+              "MOZA",
+              "MOZB",
+              "MOZD",
+              "MOZE",
+              "MOZI",
+              "MOZM",
+              "MOZO",
+              "MOZT",
+              "MOZW",
+              "MOZSL01",
+              "MOZSL02",
+              "MOZSL03",
+              "firefox-a",
+              "firefox-b",
+              "firefox-b-1",
+              "firefox-b-ab",
+              "firefox-b-1-ab",
+              "firefox-b-d",
+              "firefox-b-1-d",
+              "firefox-b-e",
+              "firefox-b-1-e",
+              "firefox-b-m",
+              "firefox-b-1-m",
+              "firefox-b-o",
+              "firefox-b-1-o",
+              "firefox-b-lm",
+              "firefox-b-1-lm",
+              "firefox-b-lg",
+              "firefox-b-huawei-h1611",
+              "firefox-b-is-oem1",
+              "firefox-b-oem1",
+              "firefox-b-oem2",
+              "firefox-b-tinno",
+              "firefox-b-pn-wt",
+              "firefox-b-pn-wt-us",
+              "ubuntu",
+              "ffab",
+              "ffcm",
+              "ffhp",
+              "ffip",
+              "ffit",
+              "ffnt",
+              "ffocus",
+              "ffos",
+              "ffsb",
+              "fpas",
+              "fpsa",
+              "ftas",
+              "ftsa",
+              "newext",
+              "monline_dg",
+              "monline_3_dg",
+              "monline_4_dg",
+              "monline_7_dg"
+            ),
+            key,
+            CONCAT(prefix, "other.scrubbed")
+          ) AS key
+        )
+      FROM
+        parsed
+    )
+      -- The SUBSTR and UNNEST logic here is copied from the prod main_summary query
+      -- see https://github.com/mozilla/bigquery-etl/blob/222c4266/sql
+      --     /moz-fx-data-shared-prod/telemetry_derived/main_summary_v4/part1.sql#L266-L273
+    SELECT
+      ARRAY_AGG(
+        STRUCT(SUBSTR(_key, 0, pos - 2) AS engine, SUBSTR(_key, pos) AS source, count)
+      )
+    FROM
+      parsed,
+      UNNEST([REPLACE(key, 'in-content.', 'in-content:')]) AS _key,
+      UNNEST([LENGTH(REGEXP_EXTRACT(_key, '.+[.].'))]) AS pos
+  )
```

Comment:
group by engine and source to avoid duplicate entries:
```suggestion
      SUBSTR(_key, 0, pos - 2) AS engine,
      SUBSTR(_key, pos) AS source,
      SUM(count) AS count,
    FROM
      parsed,
      UNNEST([REPLACE(key, 'in-content.', 'in-content:')]) AS _key,
      UNNEST([LENGTH(REGEXP_EXTRACT(_key, '.+[.].'))]) AS pos
    GROUP BY
      engine,
      source
  )
```